src/client/components/lazy-load/lazy-load.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+  <div>
+    <slot :computed="imageSource"></slot>
+  </div>
+</template>
+
+<script>
+
+export default {
+  props: ['url'],
+  data() {
+    return {
+      observer: null,
+      setSource: false
+    }
+  },
+  computed: {
+    imageSource() {
+      return this.setSource ? this.url : ''
+    }
+  },
+  mounted() {
+    const config = {
+      root: undefined,
+      rootMargin: '0px 0px 0px 0px',
```

Comment:
Nice to have this as a separate component! I would maybe move `rootMargin` and `threshold` to `props` and use the values you currently have as defaults. That way it can more easily be reused with different configurations.